conan/tools/cmake/toolchain.py

```diff
@@ -225,6 +225,20 @@ def context(self):
         return {"cppstd": cppstd, "cppstd_extensions": cppstd_extensions}
 
 
+class FindPackageDirVarsBlock(Block):
+    template = textwrap.dedent("""
+        # Help locating the find_package config files
+        {% for name in find_names %}
+        set({{ name }}_DIR "${CMAKE_BINARY_DIR}")
+        {% endfor %}
+        """)
+
+    def context(self):
+        host_req = self._conanfile.dependencies.transitive_host_requires
```

Comment:
If only necessary for OSX, maybe it should be added by default only for OSX (or Apple) systems?